### DISCUSSION TITLE
The Grand Infinite Update: Embracing Openness and Charting New Territory

### DISCUSSION THREAD

**Discussion by ntorga** [goinfinite/os#180] [Announcements]
Hello there, folks,

We've got some cracking news to share with you this month. Not only have we made some significant changes to our projects, but we've also decided to take a different path forward. And the best part? It's all good news.

## Embracing Openness

As you're aware, OS has been our darling open-source project under the EPL licence, and we're quite happy with its direction. However, as a business, we need to keep the lights on, which initially led us to plan a small monthly fee for our other project, the container management platform previously known as Control. This would have meant keeping that code behind closed doors.

But here's the thing - as the project matured, we realised what an absolute gem we had on our hands. It makes infrastructure management properly straightforward, even while in alpha stage. Dare I say, it's more intuitive than most of those fancy commercial PaaS solutions out there. All this while being self-hosted, mind you.

It would've been a terrible shame to keep this behind a paywall, putting people off before they'd even had a proper go at it. We're quite a modest operation, and while that means we're not swimming in marketing pounds, it does mean we can keep costs rather sensible. We thought to ourselves, "Why not turn this supposed weakness into something rather special?"

### The Fair Play Approach

We recently stumbled upon Fair.io. It's meant to bridge the gap between source-available proprietary code and OSI-compliant open source projects. We were rather taken with @keygen-sh's FCL approach. In essence, it lets you use it freely while preventing any unsportsmanlike competition, protecting licence mechanisms, and giving us a decent two years to sort out the business side before everything goes fully open-source.

Sure, it's not entirely open-source, but for a business-backed project, it feels like a proper balanced approach. You get to use the software without spending a penny, peek at the code, and have the peace of mind that it'll be fully open-source should anything go pear-shaped with the business.

![Young-Sheldon-GIF-by-CBS](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExbmR2N2FmNGp1b2k1Y2IwNXQ3c3NsZmQyYjBpNjRvZGVndGVxZGUzZCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/D7QvOvtAqfTti8zONJ/giphy.gif)
_Young Sheldon GIF by CBS_

### Making Ends Meet (The Gentlemanly Way)

"But how will you keep the lights on?" you might ask. Well, we've got that sorted with add-ons. Take our future backup feature, for instance - you will be able to configure your own S3 provider, but if you fancy a multi-cloud solution with generous storage and a rather splendid one-click setup, you can purchase it directly from our dashboard.

For our hosting provider friends (direct competitors), we've got a rather nice Whitelabel add-on planned that allows for re-branding and licensing the software for commercial use. Most hosting control panels charge a rather steep $20-60 per server monthly - bit rich, that. We'll likely be starting at a mere $10 monthly per server, and we'll aim to keep it accessible for the industry.

## A Spot of Rebranding

Our actual company, Infinite Cloud, has been providing managed cloud hosting in Brazil for a good decade now. We've dealt with our fair share of clunky control panels and frightfully complex Kubernetes setups, which led us to create something properly simple and swift. Initially, we planned to start a new brand called Speedia, but now that we've decided to make our main commercial software free under the FCL license, there's no point in hiding the Infinite brand.

From now on, we'll be using the Infinite brand as the umbrella for our software and hosting. This doesn't mean we'll be less open; we're embracing openness even more and keeping things simple.

The projects have had a bit of a makeover too. Keeping things straightforward, Speedia Control is now Infinite Ez (yes, that easy), and Speedia OS is now Infinite OS.

We haven't had time to update our website yet, so the /os/ and /ez/ URLs on the GitHub repositories will still return 404 for now. But we'll refactor our website to explain the projects as soon as possible. Please bear with us; our non-existent marketing team is a bit lazy, but we're focused on improving the projects.

### [Infinite OS](https://github.com/goinfinite/os) v0.1.1

We've hit double digits in versioning! We've been busy as bees refactoring the Vue.js frontend to HTMX+Alpine, and here's what we've managed:

- Completely refreshed Database Page
![databases-page](https://github.com/user-attachments/assets/60753d85-9e86-4b6d-8313-d06bd403e7bd)

- Spanking new Runtime Page
![runtimes-page](https://github.com/user-attachments/assets/11141255-1440-4852-945e-836cd9a3f472)

- Rather smart SSL Page
![ssls-page](https://github.com/user-attachments/assets/1c9d0fc0-7345-42bc-8ed4-ae5ba78388e6)

- Polished Accounts Page
![accounts-page](https://github.com/user-attachments/assets/baa17b52-6bc7-4a97-9a1b-8dca12881ab1)

- Brand new Scheduled Tasks Popover
![scheduled-tasks-modal](https://github.com/user-attachments/assets/c2d32f48-d050-444f-8242-27afc8d5a2a4)

- Rather fetching OS Banner
![os-banner](https://github.com/user-attachments/assets/5228b67f-ff2b-4ec8-ae65-350331de9410)

### [Infinite Ez](https://github.com/goinfinite/ez) v0.0.8

Bit light on updates here, as we're still tinkering with the Overview page refactor. But we've managed to roll out:

- New Scheduled Tasks Popover
![ez-scheduled-tasks-popover](https://github.com/user-attachments/assets/f41b6726-ddfc-4d1b-a2f0-aa980f32b117)

- SRI implementation for ActivityRecords
Ez now sports a resource identifier called SystemResouceIdentifier that takes a leaf out of AWS ARN's book. When you're poking about in ActivityRecords, you can pinpoint exactly which resource's changes you'd like to examine. While it might not be something you'll use in your day-to-day tinkering, it's absolutely brilliant for proper auditing work.

---

Right then, that's all for now! Do keep an eye on our updated roadmap for [OS](https://github.com/orgs/goinfinite/projects/9/views/2) and [Ez](https://github.com/orgs/goinfinite/projects/8/views/2).

Terribly grateful for your support on this journey, and we'll catch up with you in the next update! Cheers!

**Comment by polillomm** [comment]
I've worked on several projects with different proposals, target audiences, and technologies. This is the first time I feel comfortable using a solution in which I'm involved in the development. The ease of use and the simplification of containerization challenges and their management provided by the OS + Ez stack make it so robust that it could easily compete with major market players, even in its alpha version.

Even as a developer contributing to creating these solutions and knowing about future plans, I'm genuinely excited to see what this stack will bring us.

Is it fair to be a fan of your own work? 😁

**Comment by felipoantonoff** [comment]
Excelente, gostei das mudanças e ideia, desejo muito sucesso para o Projeto e Infinite Cloud, venho acompanhando há alguns anos as ideias e fico feliz que já está tomando forma.

Acho fundamental como usuário e também Empreendedor uma solução que mantenha o código-fonte aberto e transparência em boa parte do processo de desenvolvimento como vocês tem feito no Github com o Infinite OS e Infinite Ez.

Hoje uso um Painel tradicional e estou me aventurando aos poucos em Container, já rodando a mais de 1 ano um projeto nele de forma bem estável. Realmente faz falta um Painel simples de usar, com baixo consumo e baixo desempenho e também já tentei dar uma olhada em Kubernates e vamos ser sincero para a maioria dos usos é um pouco exagerado seu uso, quem sabe quando for preciso milhares de Clouds rs. 

Hoje uso Painel Opensource + Container a parte, sendo o Painel mais para o Banco, Backup, DNS, SSL, SFTP e Configurações no geral, já no Container rodo um Redmine, no qual o Host com o Nginx tem um Proxy até o Container Redmine e o Container se comunica com o nosso banco a nível do Host, agora seria lindo uma forma 100% Container ou próximo disso, talvez só o Banco fora do Container ou quem sabe ele também dentro.

A poucos anos atrás fazia tudo na raça por Shell + Scripts, curiosamente funcionava bem também, mas difícil para a maioria das pessoas e exige se manter atualizado e com boa documentação para aprender com os erros e acertos. Um ponto que acho muito importante hoje em dia é a facilidade de uso (menos horas perdidas rs), boa documentação, personalização básica, estabilidade e claro segurança, creio que estes projetos vão atender isso. 

Gostei de adotarem o nome Infinite na frente, por já ser uma marca conhecida no Brasil da Infinite Cloud e um belo nome.

Ansioso para novas novidades e para a futura versão 1.0 do Infinite Os e Ez ou a recomendada para produção, no qual pretendo usar com a Codemarket e nossos projetos, além de indicar para nossos clientes que trabalham em sua maioria com o E-commerce Opencart.
